### ISSUE TITLE
Data assigned to incorrect deviceID

### ISSUE THREAD

**Issue by zcrysler** [MotusDev/Motus-TO-DO#274]
Project 64 has the following receivers registered:

Lotek-D000426: deviceID 772, it doesn't have any deployments and was registered by the user by accident. Currently all detection data is incorrectly associated with this deviceID.

Lotek-426: deviceID 972, it has two deployments, no associated detection data, tagme() returns 0 batches although it has a deviceID so maybe there's data somewhere? The user went to the station on Friday and confirmed that the receiver is an SRX800 NOT an SRX DL

How are Lotek detections matched to deployments by serial number? Is it possible Lotek doesn't record model in the raw data so that an SRX800 and DL with number 426 are identical?

**Comment by zcrysler** [comment]
Just for reference, the 3 jobs I could see for Lotek-D000426 (https://github.com/MotusDev/Motus-TO-DO/issues/269) were found by going to the status page and searching "jobs for receiver" Lotek-D000426. It shows 138986 uploaded by user NA, jobs 109940 and 107406 uploaded by user 587, and job 144 uploaded by user 232

**Comment by Hussell** [comment]
I checked some of the files that were uploaded, and the serial number being reported in the files is D000426. So, if the receiver really is Lotek-426, then the receiver itself may be reporting the wrong ID. I don't know much about how these receivers are configured; perhaps there was a user error in entering the serial number, and it can be corrected?

**Comment by zcrysler** [comment]
The user said they visited the station on Friday and confirmed it was an SRX800 and not a DL, so I think it depends on which we trust more, the user or the Lotek files. They also said that they *think* the Lotek-D000426 was registered by mistake, but that was in 2016 and I'm not sure anyone remembers the history of this receiver

**Comment by jbrzusto** [comment]
More likely a bug in Lotek's code that generates a .DTA file from their undocumented .BIN format.
Here's an excerpt from an email from Lotek covering parts of the rules not otherwise documented
in motusServer:::getRecvModel().  Point A appears to be contradicted by the receiver above.

> Clarification:
 > 
 >  
 > 
 > A. All D type receivers manufactured after SN 000390 (June 2016) bear the SN D###### (FYI: use of D prefix became effective in June 2016.)
 > 
 >  
 > 
 > B. All M/MD type receivers are identifiable by the inclusion of the field ASP8 under heading SCAN SETTINGS (as referenced in the .DTA file. The ASP8 field is only found on M/MD variants, and therefore a unique identifier in the .DTA files.)
 > 
 >  
 > 
 > C. All D type receivers are identifiable by lack of the field ASP8 under heading SCAN SETTINGS
 > 
 >  
 > 
 >  
 > 
 > 1. Does this distinction hold for any .DTA file generated by either type of SRX800 receiver? (i.e. has this been verified from source code, rather than just from particular .DTA files?) 
 > 
 >  
 > 
 > Yes: The header segment of a data file contains a key" which identifies the type of receiver that the file came from. If the 800D key is found, it will report the antenna configuration using "AS4 - " preamble. Otherwise, it uses the "ASP8 - " designation.
 > 
 >  
 > 
 >  
 > 
 >  
 > 
 > 2. Is it independent of any settings changes a user might make, even if they don't seem sensible to you?
 > 
 >  
 > 
 > Yes, this is independent of settings and changes a user might make.
 > 
 >  
 > 
 >  
 > 
 > If you take an SRX800 M/MD Receiver and attempt to program it with SRX800D firmware, you will get a warning but it won't stop you (users CANNOT do this, this is only possible via production software). If you proceed with that, (and you're right, that would not seem sensible to me), then the receiver will at least attempt to behave like an 800D. It will try to contact hardware that it doesn't have, for example. If you managed to somehow collect some data with it (I don't think you would be able to), then the data file would identify as an SRX800D data file. Same thing is true, vice-versa.
 > 
 >  
 > 
 > If a receiver has SRX800 M/MD firmware installed, it will not allow you to upload a receiver configuration (or program live) with antenna specifications indicating an AS4. The reverse is also true. If it's an 800D, it won't accept a spec indicating an ASP8.
 > 
 >  
 > 
 >  
 > 
 >  
 > 
 > 3. And is it independent of the version of software used to download from the receiver?
 > 
 >  
 > 
 > The Host did not start using the "AS4" designation for 800D receivers in its text dump until V1.1.380.31 (November 27, 2014).
 > 
 >  
 > 
 > Receiver firmware V9.7.9 (also released November 27, 2014) was the first version that used the specific identifier flag to indicate to the Host that it was an 800D.
 > 
 >  
 > 
 > So, no. Any firmware prior to V9.7.9 (released November 27, 2014) or any Host prior to V1.1.380.31 (released November 27, 2014) is not going to help you. Both have to be later than that. Prior to those revisions, the antenna switch preamble will be "ASP8", regardless of receiver type.
 >

**Comment by Hussell** [comment]
The first deployment for this receiver started in mid Oct. 2016. The serial number is being reported as D000426, and there's no ASP8 heading in the file, just AS4.

So, if I understand it right, all the indications are that this is and SRX800D (not SRX800 and not SRX-DL), except for what the user said.

**Comment by jbrzusto** [comment]
Correcting my previous comment.  The .DTA files all give D000426 as the serial number, which does
match the above.  It is apparently an SRX800D, which is different from the SRX-DL.  But the serial number
on the unit doesn't include the "D000" portion which is why the user registered the receiver without it.

Here's an email in which this issue was discussed but we never followed up on it (email addresses
of 3rd parties depersonalized).  The above segment is part of [Lotek]'s response.

 > From: [Lotek]
 > To: john brzustowski <jbrzusto@fastmail.fm>, Stuart Mackenzie <smackenzie@bsc-eoc.org>
 > Cc: Denis Lepage <dlepage@bsc-eoc.org>, Zoe Crysler <zcrysler@bsc-eoc.org>, [Biotrack], [Lotek]
 > Subject: RE: Possible solution: Re: change to format of serial numbers in Lotek .DTA files
 > Date: Monday, 9 January 2017 19:34
 > Size: 16 KB
 > 
 > Thanks John, 
 > 
 > We will review and get back to you with a confirmation.
 > 
 > [Lotek]
 > 
 > -----Original Message-----
 > From: john brzustowski [mailto:jbrzusto@fastmail.fm] 
 > Sent: January-09-17 11:05 AM
 > To: [Lotek] ; Stuart Mackenzie
 > Cc: Denis Lepage; Zoe Crysler; [Biotrack]; [Lotek]
 > Subject: Re: Possible solution: Re: change to format of serial numbers in Lotek .DTA files
 > 
 > Hi [Lotek], Thanks for your work on this.  Is it guaranteed that this
 > distinction will hold for any .DTA files generated by either type of
 > receiver?  (i.e. has this been verified from source code, rather than
 > just from particular .DTA files?)  Is it independent of any settings
 > changes a user might make, even if they don't seem sensible to you?
 > And is it independent of the version of software used to download from
 > the receiver?
 > 
 > Sorry to be persnickety, but I'd like to avoid having to revisit this
 > once we've settled on a solution.
 > 
 > Whatever the solution ends up being, we (motus) should print up a
 > bunch of polyester stickers for all D-type receivers with serial
 > numbers less than 391, and send them to anyone we know who is using
 > those.  e.g. a sticker saying:
 > 
 >    "For use with Motus.org,
 >     Serial Number is: D000390"
 > 
 > And our code will use whatever rule we decide on to generate the
 > "correct" serial number (i.e. having a leading 'D' for those which
 > should).  That way, what users see on the receiver box will match what
 > they see in data from that box processed by motus.

**Comment by Hussell** [comment]
Ah. So the serial number written on the outside of the receiver may be in the wrong form?

**Comment by zcrysler** [comment]
User confirmed its an SRX800D. So after all this it seems like everything is working correctly and this is user error. Catherine, can you move the deployments for Lotek-426 to Lotek-D000426, and delete the Lotek-426 receiver from project 64?

**Comment by denislepage** [comment]
This is something we'll eventually need to address. Ideally, users would only have to enter the number that is visible to them, and our system would be able to do the conversion. In a case like this, however, there's no amount of validation or conversion that would work if the person chose the wrong receiver type.

We should also confirm that the data validation page looks for batches assigned to devices within the project provided during the detection data upload which are not registered to a project.

With CTT receivers, we should strongly encourage them to have a registration process initiated at the receiver end that relies on the Motus login (e.g. user pushes button, logs in with Motus and select their project).

**Comment by cjardine-bsc** [comment]
I've moved the deployments to D000426 and deleted Lotek-426. 

As to Denis's point about adding a check to the data validation process that checks for sensors not registered for the project. We do have that already (though it was only based on userID, I've now expanded it to include the projectID in sg_batches), unfortunately in this case both sensors were registered to the project so the error wasn't flagged.
